src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SchemaUpgradedHandler.cs

```diff
@@ -26,9 +26,11 @@ public Task Handle(SchemaUpgradedNotification notification, CancellationToken ca
         {
             EnsureArg.IsNotNull(notification, nameof(notification));
 
-            // TODO: Use version information to call modularized start methods (work item 75557).
-            int? version = notification.Version;
-            _sqlServerFhirModel.Start();
+            int version = notification.Version;
+            bool isFullSchemaSnapshot = notification.IsFullSchemaSnapshot;
+
+            // If it is a snapshot upgrade, then we need to run initialization for all schema versions up to the current version.
+            _sqlServerFhirModel.Initialize(version, isFullSchemaSnapshot);
```

Comment:
>isFullSchemaSnapshot [](start = 52, length = 20)

nit: I would argue about need of that variable, although compiler probably generate code with out it.

Reply:
I agree that these variables aren't needed. Thank you! Fixed in 081a167.